### PR TITLE
Add aten::log_normal_

### PR DESCRIPTION
**Pull request by hjhee**
- log_normal_

### CODE REVIEW

File: src/ATen/native/xpu/sycl/DistributionLogNormalKernel.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include <ATen/ATen.h>
+#include <ATen/AccumulateType.h>
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/DistributionLogNormalKernel.cpp:2]: remove no need to include head file

File: src/ATen/native/xpu/sycl/DistributionLogNormalKernel.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/xpu/sycl/DistributionTemplates.h>
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/DistributionLogNormalKernel.cpp:4]: remove

File: src/ATen/native/xpu/sycl/DistributionLogNormalKernel.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#include <ATen/native/xpu/sycl/Philox4x32.h>
+#include <ATen/xpu/XPUGeneratorImpl.h>
+#include <comm/DeviceProperties.h>
+#include <comm/Runtime.h>
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/DistributionLogNormalKernel.cpp:8]: remove, please check above head files are used ?

**Comment by majing921201** [comment]
add 'log_normal' to test list

File: src/ATen/native/xpu/sycl/DistributionTemplates.h [hunk 4]
```diff
@@ -698,6 +698,44 @@ void exponential_kernel(TensorIteratorBase& iter, double lambda, RNG gen) {
...
+    TensorIteratorBase& iter,
+    double mean,
+    double std,
+    RNG gen) {
+  TORCH_CHECK(std > 0.0, "log_normal_ expects std > 0.0, but found std=", std);
+  AT_DISPATCH_FLOATING_TYPES_AND2(
+      at::ScalarType::Half,
+      at::ScalarType::BFloat16,
+      iter.dtype(),
+      "log_normal__xpu_",
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/DistributionTemplates.h:727]: log_normal_xpu

File: src/ATen/native/xpu/sycl/DistributionTemplates.h [hunk 5]
```diff
@@ -698,6 +698,44 @@ void exponential_kernel(TensorIteratorBase& iter, double lambda, RNG gen) {
...
+      [&] {
+        using accscalar_t = at::acc_type_device<scalar_t, kXPU>;
+        auto mean_ = static_cast<accscalar_t>(mean);
+        auto std_ = static_cast<accscalar_t>(std);
+        // define functor to multiply std and add mean
+        LogNormalFunctor<scalar_t, accscalar_t> log_normal_functor(mean_, std_);
+        uniform_and_transform<scalar_t, accscalar_t, rand4_engine_calls>(
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/DistributionTemplates.h:734]: You should use `normal_and_transform` as CUDA does

**Review by xytintel — CHANGES_REQUESTED** [review]

**Comment by xytintel** [comment]
image
image

**Review by xytintel — APPROVED** [review]

**Pull request merged**